### PR TITLE
Use uuid v7 when adding a space, change method signature

### PR DESCRIPTION
**Pull request by nvinuesa**
_Note: This PR is a prelude to the the migration work being done. And this is also needed for the add of the ReloadSpaces functionality._

This patch updates the google/uuid library so we can make use of the new UUID V7 support. More specificaly, when adding a space we check if the provided space contains an ID, if it doesn't we generate a new V7 UUID for it.

We also changed the signature on the AddSpace method (by replacing the individual parameters by a `network.SpaceInfo` struct) of the service layer to do this.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

This method is still not wired, so simply running the unit test should suffice:
```
TEST_PACKAGES="./domain/network/... -count=1 -race -check.vv" make run-go-tests
```

## Links

**Jira card:** JUJU-4835

### CODE REVIEW

File: domain/network/service/space_test.go [hunk 1]
```diff
@@ -90,7 +107,12 @@ func (s *spaceSuite) TestAddSpace(c *gc.C) {
...
 
-	returnedUUID, err := NewService(s.st, s.logger).AddSpace(context.Background(), "space0", network.Id("provider-id"), []string{})
+	returnedUUID, err := NewService(s.st, s.logger).AddSpace(
+		context.Background(),
+		network.SpaceInfo{
+			Name:       "space0",
+			ProviderId: "provider-id",
+		})
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(returnedUUID.String(), gc.Equals, expectedUUID)
```
Comment (Aflynn50) [inline domain/network/service/space_test.go:117]: Does this test actually test that a new uuid is generated for spaces that don't have one? I guess it checks that the uuid is the same as the one going into `state.AddSpace`, but that may also be blank.

**Review by Aflynn50 — APPROVED** [review]
Looks good, though I'm not sure that any of the tests actually check that you generate a new UUID if you leave it blank.

File: domain/network/service/space_test.go (hunk 1 above)
Comment (nvinuesa) [reply domain/network/service/space_test.go]: Indeed, this test does not mock the uuid generation so we cannot test that the method is called. However this would not be very practical (we are only generating one field).
I did add a check on the test to make sure that the generated ID is not empty. The test already provides an empty ID as input, and makes sure that the state method receives a non-empty ID. This is enough IMO.
